### PR TITLE
Buttons

### PR DESCRIPTION
**Pull request by martypowell**
Addresses #37 

- Primary Buttons
- Secondary Buttons
- Link Buttons
- Icon Links

Adds animation to hover for buttons and links. Added web fonts to display font awesome.

Does not add navigation buttons, that will be handled as part of the navigation build

### CODE REVIEW

**Comment by martypowell** [comment]
@martypowell please add examples for the button usage to the docs.

File: src/styles/partials/components/_button.scss
Comment (martypowell) [inline src/styles/partials/components/_button.scss]: these can be simlified a bit.

Comment (martypowell) [reply src/styles/partials/components/_button.scss]: simplified as much as possible

File: src/components/basic/IconLink.js
Comment (martypowell) [inline src/components/basic/IconLink.js]: we should probably just control this through a class, in order to simplify this. Be sure to update the docs.

Comment (martypowell) [reply src/components/basic/IconLink.js]: this is done.

File: src/styles/partials/components/_link.scss
Comment (martypowell) [inline src/styles/partials/components/_link.scss:69]: this looks like a good case for a mixin

**Review by danfox01 — CHANGES_REQUESTED** [review]
@martypowell looks great, except for a few things:

- Hover state of `dg_button-link` in a dark section is incorrect. The background color should not change on hover, instead the link color should change.
- Font weight of `dg_button-link` should be 600 (semibold) not 700 (bold)
- I noticed we're not currently pulling the italic version of Crimson. We need to add this for true italics.

FYI @sdurphy

File: src/styles/partials/components/_link.scss
Comment (martypowell) [reply src/styles/partials/components/_link.scss:69]: done in latest changes.

**Comment by martypowell** [comment]
> @martypowell looks great, except for a few things:
> 
> * Hover state of `dg_button-link` in a dark section is incorrect. The background color should not change on hover, instead the link color should change.
> * Font weight of `dg_button-link` should be 600 (semibold) not 700 (bold)
> * I noticed we're not currently pulling the italic version of Crimson. We need to add this for true italics.
> 
> FYI @sdurphy

Hey @sdurphy @danfox01 did these get addressed?

**Comment by danfox01** [comment]
Per testing by @jdomasky, a few inconsistencies and variation from the comps:

- For `dg_icon-link` in a dark section, the hover color should be `blue-lightest` not `blue-vibrant`.
- Most of the active state examples also respond to hover. The exception seems to be `dg_icon-link` and `dg_icon-link--small`, which do not show a hover state. Seems weird to me that an example of an active state would have a hover, so I'm not sure which approach is correct. For reference, the [active button examples in the USDS style guid](https://designsystem.digital.gov/components/button/)e do not respond to hover. Regardless, our examples should be consistent.

It's also worth noting that the buttons sizes may need to be tweaked for mobile in future iterations. This might be related to responsive typography.

File: src/styles/partials/components/_link.scss
Comment (theilman7) [inline src/styles/partials/components/_link.scss:25]: This is more of a question. I'm wondering if we should include values for line-height and other CSS typographic focused variables in the _fonts.scss file. IE. $line-height-lg, $line-height-sm etc.

File: src/styles/partials/base/_button-reset.scss
Comment (theilman7) [inline src/styles/partials/base/_button-reset.scss:11]: For consistency and for cleaning up the markup, I would add the background property before border so we can maintain the alphabetic order of css properties.

File: src/styles/partials/components/_button.scss
Comment (theilman7) [inline src/styles/partials/components/_button.scss:50]: For consistency, move the border-color property below the background property.

File: src/styles/partials/components/_label.scss
Comment (theilman7) [inline src/styles/partials/components/_label.scss:5]: These properties should be in the following order for consistency: font-size, font-weight, margin-top.

File: src/styles/partials/base/_button-reset.scss
Comment (martypowell) [reply src/styles/partials/base/_button-reset.scss:11]: i copied this from somewhere, else, but I'm good wtih this comment.

File: src/styles/partials/components/_button.scss
Comment (martypowell) [reply src/styles/partials/components/_button.scss:50]: good catch, done.

File: src/styles/partials/components/_label.scss
Comment (martypowell) [reply src/styles/partials/components/_label.scss:5]: adjusted.

File: src/styles/partials/components/_link.scss
Comment (martypowell) [reply src/styles/partials/components/_link.scss:25]: I don't have a strong knowledge of how to effectively use line height, it's always more have been a case by case basis for me personally. If i you have a good handle on this, then I'd defer to you. But I'm always in favor of creating some more rules around this.

File: src/styles/partials/base/_button-reset.scss
Comment (martypowell) [reply src/styles/partials/base/_button-reset.scss:11]: adjusted any no commented proprties

**Comment by martypowell** [comment]
> Per testing by @jdomasky, a few inconsistencies and variation from the comps:
> 
> * For `dg_icon-link` in a dark section, the hover color should be `blue-lightest` not `blue-vibrant`.
> * Most of the active state examples also respond to hover. The exception seems to be `dg_icon-link` and `dg_icon-link--small`, which do not show a hover state. Seems weird to me that an example of an active state would have a hover, so I'm not sure which approach is correct. For reference, the [active button examples in the USDS style guid](https://designsystem.digital.gov/components/button/)e do not respond to hover. Regardless, our examples should be consistent.
> 
> It's also worth noting that the buttons sizes may need to be tweaked for mobile in future iterations. This might be related to responsive typography.

Looking now. I also think i got the clasess wrong, looking now. Since doing this pr I realized I should have done one button set a time :disappointed:

**Comment by martypowell** [comment]
> > Per testing by @jdomasky, a few inconsistencies and variation from the comps:
> > 
> > * For `dg_icon-link` in a dark section, the hover color should be `blue-lightest` not `blue-vibrant`.
> > * Most of the active state examples also respond to hover. The exception seems to be `dg_icon-link` and `dg_icon-link--small`, which do not show a hover state. Seems weird to me that an example of an active state would have a hover, so I'm not sure which approach is correct. For reference, the [active button examples in the USDS style guid](https://designsystem.digital.gov/components/button/)e do not respond to hover. Regardless, our examples should be consistent.
> > 
> > It's also worth noting that the buttons sizes may need to be tweaked for mobile in future iterations. This might be related to responsive typography.
> 
> Looking now. I also think i got the classess wrong, looking now. Since doing this pr I realized I should have done one button set a time :/

@danfox01 I think all issues have been addressed, please review.

**Comment by danfox01** [comment]
Per conversation with @martypowell final review of this issue is on hold until later this week. When we resume, here are retest results from @jdomasky:

> 1. The hover class was removed from all of the Active buttons except for one: Text Button in a Dark Section.
> 2. In the Decorative Links in a Dark Section (both Large and Small), the Disabled buttons have a hover class configured. I believe this is a new bug. I don't remember observing hover on these buttons when we tested yesterday.

**Comment by martypowell** [comment]
> Per conversation with @martypowell final review of this issue is on hold until later this week. When we resume, here are retest results from @jdomasky:
> 
> > 1. The hover class was removed from all of the Active buttons except for one: Text Button in a Dark Section.
> > 2. In the Decorative Links in a Dark Section (both Large and Small), the Disabled buttons have a hover class configured. I believe this is a new bug. I don't remember observing hover on these buttons when we tested yesterday.

Both valid bugs and addressed @jdomasky ready for another look.

fyi @danfox01

**Comment by jdomasky** [comment]
@martypowell @danfox1 Taking another look.

**Comment by jdomasky** [comment]
@martypowell @danfox1 The issues in the retest results from this morning have been resolved. However, there's still an issue with this component:

- In the Secondary Button set and the Text Button in a Dark Section, the Disabled buttons have a hover class configured. I'm not sure if this was overlooked during retesting or if this issue was just introduced.
**Recommendation:** Remove the hover class from these two Disabled buttons.

**Comment by martypowell** [comment]
> @martypowell @danfox1 The issues in the retest results from this morning have been resolved. However, there's still an issue with this component:
> 
> * In the Secondary Button set and the Text Button in a Dark Section, the Disabled buttons have a hover class configured. I'm not sure if this was overlooked during retesting or if this issue was just introduced.
>   **Recommendation:** Remove the hover class from these two Disabled buttons.

@jdomasky this is a bug, and was just overlooked, fixing now.

File: src/styles/partials/components/_button.scss
Comment (martypowell) [reply src/styles/partials/components/_button.scss:50]: @theilman7 ready for another look

File: src/styles/partials/components/_label.scss
Comment (martypowell) [reply src/styles/partials/components/_label.scss:5]: @theilman7 ready for another look

File: src/styles/partials/components/_link.scss
Comment (martypowell) [reply src/styles/partials/components/_link.scss:25]: > > @martypowell @danfox1 The issues in the retest results from this morning have been resolved. However, there's still an issue with this component:
> > 
> > * In the Secondary Button set and the Text Button in a Dark Section, the Disabled buttons have a hover class configured. I'm not sure if this was overlooked during retesting or if this issue was just introduced.
> >   **Recommendation:** Remove the hover class from these two Disabled buttons.
> 
> @jdomasky this is a bug, and was just overlooked, fixing now.

@jdomasky this is ready for another look.

**Comment by danfox01** [comment]
@martypowell , this is good to merge.

File: src/styles/partials/components/_link.scss
Comment (jdomasky) [reply src/styles/partials/components/_link.scss:25]: @martypowell The buttons are good to go. All testing issues have been resolved.

**Pull request merged**
